homeassistant/components/androidtv/__init__.py

```diff
@@ -1 +1,221 @@
+
+    else:
+        # Use "pure-python-adb" (communicate with ADB server)
+        signer = None
+        adb_log = f"using ADB server at {config[CONF_ADB_SERVER_IP]}:{config[CONF_ADB_SERVER_PORT]}"
+
+    return adbkey, signer, adb_log
+
+
+def validate_state_det_rules(state_det_rules):
+    """Validate a string that contain state detection rules and return a dict."""
+    if not state_det_rules:
+        return None
+
+    try:
+        json_rules = json.loads(state_det_rules)
+    except ValueError:
+        _LOGGER.warning("Error loading state detection rules")
+        return None
+    try:
+        state_detection_rules_validator(json_rules, ValueError)
+    except ValueError as exc:
+        _LOGGER.warning("Invalid state detection rules: %s", exc)
+        return None
+    return json_rules
+
+
+async def async_connect_androidtv(
+    hass, config, *, state_detection_rules=None, timeout=30.0
+):
+    """Connect to Android device."""
+    address = f"{config[CONF_HOST]}:{config[CONF_PORT]}"
+
+    adbkey, signer, adb_log = await hass.async_add_executor_job(
+        _setup_androidtv, hass, config
+    )
+
+    aftv = await setup(
+        config[CONF_HOST],
+        config[CONF_PORT],
+        adbkey,
+        config.get(CONF_ADB_SERVER_IP),
+        config.get(CONF_ADB_SERVER_PORT, DEFAULT_ADB_SERVER_PORT),
+        state_detection_rules,
+        config[CONF_DEVICE_CLASS],
+        timeout,
+        signer,
+    )
+
+    if not aftv.available:
+        # Determine the name that will be used for the device in the log
+        if config[CONF_DEVICE_CLASS] == DEVICE_ANDROIDTV:
+            device_name = "Android TV device"
+        elif config[CONF_DEVICE_CLASS] == DEVICE_FIRETV:
+            device_name = "Fire TV device"
+        else:
+            device_name = "Android TV / Fire TV device"
+
+        _LOGGER.warning(
+            "Could not connect to %s at %s %s", device_name, address, adb_log
+        )
+        return None
+
+    return aftv
+
+
+def _migrate_aftv_entity(hass, aftv, entry_unique_id):
+    """Migrate a entity to new unique id."""
+    entity_reg = er.async_get(hass)
+
+    entity_unique_id = entry_unique_id
+    if entity_reg.async_get_entity_id(MP_DOMAIN, DOMAIN, entity_unique_id):
+        # entity already exist, nothing to do
+        return
+
+    old_unique_id = aftv.device_properties.get(PROP_SERIALNO)
+    if not old_unique_id:
+        # serial no not found, exit
+        return
+
+    migr_entity = entity_reg.async_get_entity_id(MP_DOMAIN, DOMAIN, old_unique_id)
+    if not migr_entity:
+        # old entity not found, exit
+        return
+
+    try:
+        entity_reg.async_update_entity(migr_entity, new_unique_id=entity_unique_id)
+    except ValueError as exp:
+        _LOGGER.warning("Migration of old entity failed: %s", exp)
+
+
+async def async_setup(hass, config):
+    """Set up the Android TV integration."""
+    return True
+
```

Comment:
```suggestion
```

We can drop this

Reply:
As I know we cannot drop this because it is required to run `async_setup_platform` that is still there for the import flow.

Reply:
It shouldn't be needed if you have `async_setup_entry` defined after https://github.com/home-assistant/core/pull/48381

Reply:
If I have a yaml configuration and I run Home Assistant without this block I receive this error:

```
[homeassistant.setup] Setup failed for androidtv: Integration 'androidtv' did not return boolean if setup was successful. Disabling component.
```

and obviously import fail.

Reply:
When was the last time you merged dev into your branch for this PR? Maybe that has something to do with it?

Reply:
It is merged with last available dev release and I rebased this PR just yesterday